receiver/mysqlreceiver/config.go

```diff
@@ -0,0 +1,47 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysqlreceiver
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	Username                                string `mapstructure:"username"`
+	Password                                string `mapstructure:"password"`
+	Database                                string `mapstructure:"database"`
+	Endpoint                                string `mapstructure:"endpoint"`
+}
+
+// Errors for missing required config parameters.
+const (
+	ErrNoUsername = "invalid config: missing username"
+	ErrNoPassword = "invalid config: missing password" // #nosec G101 - not hardcoded credentials
+)
+
+func (cfg *Config) Validate() error {
+	var errs []error
+	if cfg.Username == "" {
+		errs = append(errs, errors.New(ErrNoUsername))
+	}
+	if cfg.Password == "" {
```

Comment:
Should we allow password to be empty? For example, administrators may configure MySQL to allow a certain user to log in only from the local machine, but may not require a password https://dev.mysql.com/doc/mysql-secure-deployment-guide/8.0/en/secure-deployment-configure-authentication.html.

Reply:
We should support this. I'll have someone on my team see whether this is something that can be added quickly, and if so, will include it in a subsequent PRs.

I think this is not the most common use case though, so I'd prefer to consider this non-blocking. Please challenge me on this if you disagree.

Reply:
In the interest of moving things forward, I'd prefer that we relax the constraint later, when we can include an integration test to demonstrate the functionality. This can likely be done in #5456.